index.js

```diff
@@ -503,28 +503,37 @@ module.exports.removeTags = function(filepath, fn) {
     return removeTagsSync(filepath)
 }
 
-function makePromise(fn) {
+function makeSwapParameters(fn) {
+    return (a, b) => fn(b, a)
+}
+
+// The reorderParameter is a workaround because the callback function
+// does not have a consistent interface between all the API functions.
+// Ideally, all the functions should align with the promise style and
+// always have the result first and the error second.
+// Changing this would break the current public API.
+// This could be changed internally and swap the parameter in a light
+// wrapper when creating the public interface and then remove in a
+// version 1.0 later with an API breaking change.

```

Comment:
This big comment is cluttering a bit the code and could be in the commit message instead, let me know what you think.

Reply:
I'd leave it in for now because there are multiple comments in the code about v1 changes, we could extract everything at once into an issue or something like that later.

Reply:
OK, thanks.